### ISSUE TITLE
Heading does not render with dependsOn

### ISSUE THREAD

**Issue by KasparRosin** [laravel/nova-issues#4811]
- Nova Version: 4.14.1

### Description:
In the latest release, it seems that heading field no longer works with dependsOn.
The field will render for a split second and then disappear as in the visible value was set to false.

### Detailed steps to reproduce the issue on a fresh Nova installation:
```php
Select::make('Manual', 'manual')
                ->options([false => 'Manual', true => 'Automatic'])
                ->nullable(false)
                ->displayUsingLabels()
                ->resolveUsing(fn ($val) => intval($val) ?? 0)
                ->default(false),

Heading::make('Manual')
                ->dependsOn(['manual'], function ($field, $request, $data) {
                    $field->withMeta(['value' => $data->manual ? 'Manual' : 'Automatic']);
                }),
```

**Comment by davidhemphill** [comment]
Your `Heading` field uses the same underlying attribute as the `Select` field ("manual"). The attributes should be different so the listeners know what to check for.

**Comment by recycledbeans** [comment]
I had a similar issue, but the label (and underlying attribute) was not the same as the field the Heading was depending on. In my case, any time the Text field I was depending on changed, the dependent Heading field was removed.

I was able to fix this by explicitly passing an attribute value in the second param of the Heading::make() method in addition to the name.

**Comment by KasparRosin** [comment]
@davidhemphill Sorry, for the example I shortened the name field.
The actual name field was `Insert manual fields` and `Select automatic value"

**Comment by davidhemphill** [comment]
Found the issue. The way `Heading` is currently constructed, it does not have a `name` property (and thus no underlying attribute to track). Adding the second `attribute` parameter fixes this. Adding the second attribute fixes the dependency issue but introduces another one because Nova attempts to fill the `Heading` field's attribute when saving the model, causing an error. 

I've fixed this in the latest version, and we'll push out a patch release today.
